src/screens/map-view-test.tsx

```diff
@@ -0,0 +1,199 @@
+            setInputToRTK(datas);
+        }
+    }, [regionChange]);
+
+    // END of NEW way
+
+    return (
+        <Box style={styles.container}>
+            {location ? (
+                <MapView
+                    style={styles.map}
+                    onRegionChangeComplete={(e) => {
+                        // console.log(e);
+                        setRegionChange(e);
+                    }}
+                    initialRegion={{
+                        latitude: location[0],
+                        longitude: location[1],
+                        latitudeDelta: 0.0922,
+                        longitudeDelta: 0.0421,
+                    }}>
+                    <Marker
+                        title="You Are Here"
+                        coordinate={{ latitude: location[0], longitude: location[1] }}></Marker>
+                    {product &&
+                        product.map((prod) => {
+                            return (
+                                <Marker
+                                    onPress={() => {
+                                        if (prod.id) {
+                                            navigation.navigate('Product', {
+                                                id: prod.id,
+                                            });
+                                        }
+                                    }}
+                                    key={prod.id}
+                                    title="Event"
+                                    pinColor="gold"
+                                    coordinate={{
+                                        latitude: prod.lat,
+                                        longitude: prod.long,
+                                    }}></Marker>
+                            );
+                        })}
+                    <Circle
+                        center={{ latitude: location[0], longitude: location[1] }}
+                        radius={1000 * distance}
+                    />
+                    {regionChange && (
+                        <Circle
+                            center={{
+                                latitude: regionChange.latitude,
+                                longitude: regionChange.longitude,
+                            }}
+                            radius={regionChange.latitudeDelta * 500 * 110.9472}
+                        />
+                    )}
+                </MapView>
+            ) : (
+                <ActivityIndicator color="#36d7b7" />
+            )}
+
+            <Slider
+                w="3/4"
+                maxW="300"
+                defaultValue={distance}
+                minValue={10}
+                maxValue={2000}
+                accessibilityLabel="hello world"
+                step={100}
+                onChangeEnd={(v) => {
+                    setDistance(v);
+                }}
+                onChange={(v) => {
+                    setDistanceLabel(v);
+                }}>
```

Comment:
just to clean up code, you actually do not need the extra brackets, can do the following:
```
onChange{(v) => setDistance(v)}
```